streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -218,10 +218,10 @@ public void processInOrder(final KIn key, final VIn value, final long inputRecor
                         if (windowMaxRecordTimestamp < inputRecordTimestamp) {
                             previousRecordTimestamp = windowMaxRecordTimestamp;
                         }
-                        updateWindowAndForward(windowBeingProcessed.key.window(), windowBeingProcessed.value, key, value, closeTime, inputRecordTimestamp);
+                        updateWindowAndForward(windowBeingProcessed.key.window(), windowBeingProcessed.value, record, closeTime, inputRecordTimestamp);
```

Comment:
Likewise here, no need to pass along `inputRecordTimestamp` when the record also has the timestamp, and also in a few other place in this class.